### DISCUSSION TITLE
TypeError: __init__() takes 1 positional argument but 2 were given

### DISCUSSION THREAD

**Discussion by Mandrohamburger** [mrdbourke/pytorch-deep-learning#1041] [Q&A]
I am 23:04:26 into the video and have come across the title error when writing the following code,
```
img_batch, label_batch = next(iter(train_dataloader_simple))

img_batch.shape, label_batch.shape
```

I believe there is some error in the next() function as the error specifies
```
Cell In[253], line 1
----> 1 img_batch, label_batch = next(iter(train_dataloader_simple))
      3 img_batch.shape, label_batch.shape

File ~/Library/Python/3.9/lib/python/site-packages/torch/utils/data/dataloader.py:630, in _BaseDataLoaderIter.__next__(self)
    627 if self._sampler_iter is None:
    628     # TODO(https://github.com/pytorch/pytorch/issues/76750)
    629     self._reset()  # type: ignore[call-arg]
--> 630 data = self._next_data()
    631 self._num_yielded += 1
    632 if self._dataset_kind == _DatasetKind.Iterable and \
    633         self._IterableDataset_len_called is not None and \
    634         self._num_yielded > self._IterableDataset_len_called:
```
Any help is greatly appreciated!

**Comment by LuluW8071** [comment]
Next function is inside the torch library utils/data/dataloader.py file

The error is probably somewhere else on ur code 
Paste the full error

**Comment by mrdbourke** [comment]
Hey @Mandrohamburger,

Have you tried inspecting what `train_dataloader_simple` contains manually?

You may need to see print out what it looks like/check out how it's constructed before printing `next(iter(train_dataloader_simple))`, looks like it may have been created incorrectly (though I'm not 100% sure on this).

Do you have a small reproducible version of your code that shows the error? 

That would really help.
